### PR TITLE
[lldb][NFC] Move few static helpers to the class Socket

### PR DESCRIPTION
**Pull request by slydiman**
It is just an optimization.

### CODE REVIEW

**Review by labath — CHANGES_REQUESTED** [review]
I don't think this is a good change.

**Comment by slydiman** [comment]
> I don't think this is a good change.

What exactly is wrong here?

Is that code good duplicated in Socket.cpp and TCPSocket.cpp?
```
#if defined(_WIN32)
  bool success = closesocket(m_socket) == 0;
#else
  bool success = ::close(m_socket) == 0;
#endif
```

SetLastError(Status &error) is placed in Socket, but GetLastError() in TCPSocket. Is it ok?

**Comment by labath** [comment]
Moving the common API (deduplicating) into the `Socket` class is fine. Making it public is not. There shouldn't be a reason to make e.g. `GetLastError` public, because noone should be making raw socket calls.

File: lldb/include/lldb/Host/Socket.h [hunk 1]
```diff
@@ -34,9 +34,11 @@ namespace lldb_private {
 #if defined(_WIN32)
 typedef SOCKET NativeSocket;
 typedef lldb::pipe_t shared_fd_t;
+typedef const char *set_socket_option_arg_type;
```
Comment (labath) [inline lldb/include/lldb/Host/Socket.h:37]: Instead of exposing this quirk, it would be better to have a static version of `Socket::SetOption` to hide it.

Comment (slydiman) [reply lldb/include/lldb/Host/Socket.h]: Updated. Fixed also a typo
`set`_socket_option_arg_type option_value_p =
      reinterpret_cast<`get`_socket_option_arg_type>(&option_value);

**Review by labath — APPROVED** [review]
cool. thanks.

**Pull request merged**